app/controllers/v1/credentials_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module V1
+  class CredentialsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :set_app
+    before_action :set_credential
+
+    def regenerate_token
+      if @cred&.update(token: secure_token)
```

Comment:
At this point, @credential shouldn't be null.

Another thing, we could have a `regenerate_token!` method in the `credential model`, In this way, we avoid generating the token from [here](https://github.com/gbrlmrllo/chatops_builder_api/pull/7/files#diff-acb03f978f7c1ce520df0ede6c7b8d1cR28).